src/v3/src/util/widgetHooks.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { IdxStatus, IdxTransaction } from '@okta/okta-auth-js';
+
+import { HookFunction, HooksOptions, HookType } from '../../../types';
+import { getFormNameForTransaction } from './getEventContext';
+
+/* eslint-disable no-await-in-loop, no-restricted-syntax */
+export async function executeWidgetHooks(
+  hookType: HookType,
+  hooks?: HooksOptions,
```

Comment:
Curious if `hooks` can be kept as a singleton in this file instead of using a `ref` in the `Widget/index.tsx` file and needing to pass the ref around, awkwardly needing to access `ref.current`, etc.

Since it's a `ref` we already are implying it does not affect component rendering so we could potentially simplify by using a singleton. I might also suggest using `Map`s here instead of arrays if we can make that move.